internal/api_server/server.go

```diff
@@ -68,6 +68,12 @@ func oapiErrorHandler(w http.ResponseWriter, message string, statusCode int) {
 	http.Error(w, fmt.Sprintf("API Error: %s", message), statusCode)
 }
 
+const (
+	maxUrlLength   = 2000             // Limit the length of URLs
+	maxNumHeaders  = 10               // Limit the number of headers in a request
```

Comment:
```suggestion
	maxNumHeaders  = 10               // Limit the number of headers in a request
```

Why so low?, If we have proxies of any sort in the middle, those can include extra and unexpected headers.

Reply:
Increased the default to 32 and made all of these configurable too.